src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -43,18 +44,22 @@ internal sealed class SonarErrorListDataSource :
         IDisposable
     {
         private readonly IFileRenamesEventSource fileRenamesEventSource;
+        private readonly IAnalysisIssueSelectionService selectionService;
         private readonly ISet<ITableDataSink> sinks = new HashSet<ITableDataSink>();
         private readonly ISet<IIssuesSnapshotFactory> factories = new HashSet<IIssuesSnapshotFactory>();
 
         [ImportingConstructor]
-        internal SonarErrorListDataSource(ITableManagerProvider tableManagerProvider, IFileRenamesEventSource fileRenamesEventSource)
+        internal SonarErrorListDataSource(ITableManagerProvider tableManagerProvider, 
+            IFileRenamesEventSource fileRenamesEventSource,
+            IAnalysisIssueSelectionService selectionService)
```

Comment:
This doesn't feel right. The responsibility of the data source class is to handle the integration with the VS Error List. It shouldn't need to know about the selection service. If we need more complicated logic to synchronize the selection service/panel/error list it should be handled elsewhere.

Reply:
I agree. Currently, the analysis results are reported directly to the Snapshot classes. That should've been the place to synchronize the selection service, however I felt that it was too big of a change to do in the last minute. I agree though; I also had to copy logic into the panel, which I really didn't want to do. It was just to minimize the scope of the changes.

Reply:
Btw, I thought about doing a hacky solution and in the lightbulb just selecting the item in error list. However, the error list might not yet be loaded (and the lightbulb is), and also the error list might have a filter, in which case there is nothing to select.

Reply:
If you're knowingly doing something hacky it's worth capturing that information somehow, either in comments in the code or the PR, otherwise it's very hard later to work out later if there was a good reason for something that looks wrong.

I agree about the underlying problem, but I don't think a different state service is the answer. The error list data source is still the single source of truth about the set of active issues and it raises an event when the set of issues changes.
The behaviour regarding the visualisation of the current selection is a UI issue that should be handled at a higher level.